pkg/tsdb/elasticsearch/time_series_query.go

```diff
@@ -179,6 +179,49 @@ func (e *timeSeriesQuery) processQuery(q *Query, ms *es.MultiSearchRequestBuilde
 	return nil
 }
 
+// Casts values to int when required by Elastic's query DSL
+func (metricAggregation MetricAgg) castSettings() map[string]interface{} {
```

Comment:
Naming is hard.
This converts a query model settings to an ES DSL compatible object by casting required values without altering the source variable.

`getDSLCompatibleValues`, `getCastedValues`, `formatSettings`? 🤔

Reply:
the current `generateSettingsForDSL` maybe?